### PR TITLE
Timeline design change feature

### PR DESCRIPTION
**Pull request by jrvboesch**
added a new prop to timeline called mode where you can send 'alternate' to change the timeline design and 'right' to flip horizontally.

### CODE REVIEW

**Comment by afc163** [comment]
The demo seems to overflow, cloud you optimize it?

image

File: components/timeline/Timeline.tsx [hunk 1]
```diff
@@ -11,27 +11,31 @@ export interface TimelineProps {
...
+      mode,
       ...restProps
     } = this.props;
     const pendingNode = typeof pending === 'boolean' ? null : pending;
     const classString = classNames(prefixCls, {
       [`${prefixCls}-pending`]: !!pending,
       [`${prefixCls}-reverse`]: !!reverse,
-    }, className);
+    }, className,
+    mode);
```
Comment (afc163) [inline components/timeline/Timeline.tsx]: className `${prefixCls}-${mode}` is prefered.

File: components/timeline/demo/alternate.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+order: 3
+title:
+  zh-CN: 基本用法
```
Comment (afc163) [inline components/timeline/demo/alternate.md]: You can use `交替展现` as chinese description.

File: components/timeline/demo/alternate.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  en-US: Alternate
+---
+
+## zh-CN
+
+基本的时间轴。
```
Comment (afc163) [inline components/timeline/demo/alternate.md]: You can use `内容在时间轴两侧轮流出现。` as chinese description.

File: components/timeline/demo/right.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+order: 4
+title:
+  zh-CN: 自定义时间轴点
```
Comment (afc163) [inline components/timeline/demo/right.md]: You can use `右侧时间轴点` as chinese title.

File: components/timeline/demo/right.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  en-US: Right alternate
+---
+
+## zh-CN
+
+可以设置为图标或其他自定义元素。
```
Comment (afc163) [inline components/timeline/demo/right.md]: You can use `时间轴点可以在内容的右边。` as chinese title.

File: components/timeline/__tests__/__snapshots__/demo.test.js.snap [hunk 6]
```diff
@@ -109,7 +109,7 @@ exports[`renders ./components/timeline/demo/alternate.md correctly 1`] = `
 
 exports[`renders ./components/timeline/demo/basic.md correctly 1`] = `
 <ul
-  class="ant-timeline"
+  class="ant-timeline ant-timeline-"
```
Comment (afc163) [inline components/timeline/__tests__/__snapshots__/demo.test.js.snap]: oops

File: components/timeline/index.zh-CN.md [hunk 7]
```diff
@@ -32,6 +32,7 @@ title: Timeline
 | pending | 指定最后一个幽灵节点是否存在或内容 | boolean\|string\|ReactNode | false |
 | pendingDot | 当最后一个幽灵节点存在時，指定其时间图点 | \|string\|ReactNode | `<Icon type="loading" />` |
 | reverse | 节点排序 | boolean | false |
+| mode | By sending `alternate` the timeline will distribute the nodes to the left and right.  | `standard` \| `alternate` | `standard` |
```
Comment (afc163) [inline components/timeline/index.zh-CN.md]: Two standard should be `left` and `right`.

Comment (afc163) [reply components/timeline/index.zh-CN.md]: And chinese goes to 

```
通过设置 `mode` 可以改变时间轴和内容的相对位置
```

Comment (jrvboesch) [reply components/timeline/index.zh-CN.md]: wow sorry, bad copy & paste.

**Comment by afc163** [comment]
@jrvboesch Nice job!!! 😀

**Pull request merged**